File: experiments/source/python/quote_filter.py
import loom


@loom.entry
def quote_filter(threshold: float, bid: loom.Tensor1, ask: loom.Tensor1) -> loom.Tensor1:
    def filter_quote(bi: float, ai: float) -> float:
        spread = ai - bi
        return spread if spread > threshold else 0.0

    return loom.Tensor1.map2(filter_quote, bid, ask)

File: experiments/source/cpp/quote_filter.cpp
#include <loom/loom.hpp>

LOOM_ENTRY loom::Tensor1 quote_filter(float threshold, loom::Tensor1 bid, loom::Tensor1 ask) {
  auto filter_quote = [=](float bi, float ai) -> float {
    auto spread = ai - bi;
    return spread > threshold ? spread : 0.0f;
  };
  return loom::Tensor1::map2(filter_quote, bid, ask);
}
